src/App.tsx

```diff
@@ -1,73 +1,122 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
+
+const delay = 300;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const people: Person[] = peopleFromServer;
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(true);
+  const [chosenPerson, setChosenPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
+    setChosenPerson(null);
+    setIsDropdownActive(true);
+  };
+
+  const handlePersonClick = (slug: string) => {
+    const foundedPerson = people.find(person => person.slug === slug) || null;
+
+    setChosenPerson(foundedPerson);
+    setIsDropdownActive(false);
+  };
+
+  const handleInputClick = () => {
+    setIsDropdownActive(true);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, people]);
+
+  // test
+  const handleInputBlur = () => {
+    setTimeout(() => setIsDropdownActive(false), 100);
```

Comment:
Using `setTimeout` in `handleInputBlur` can lead to unexpected behavior if the component unmounts before the timeout completes. Consider using a cleanup function to clear the timeout if necessary.